src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs

```diff
@@ -26,52 +26,70 @@ public static class ScriptJwtBearerExtensions
-                                    {
-                                        o.TokenValidationParameters = CreateTokenValidationParameters();
-                                    }
+                        // Temporary: Tactical fix to address specialization issues. This should likely be moved to a token validator
+                        // TODO: DI (FACAVAL) This will be fixed once the permanent fix is in place
+                        if (_specialized == 0 && !SystemEnvironment.Instance.IsPlaceholderModeEnabled() && Interlocked.CompareExchange(ref _specialized, 1, 0) == 0)
+                        {
+                            o.TokenValidationParameters = CreateTokenValidationParameters(GetValidAudiences());
+                        }
 
-                                    return Task.CompletedTask;
-                                },
-                                OnTokenValidated = c =>
-                                {
-                                    c.Principal.AddIdentity(new ClaimsIdentity(new Claim[]
-                                    {
-                                        new Claim(SecurityConstants.AuthLevelClaimType, AuthorizationLevel.Admin.ToString())
-                                    }));
+                        return Task.CompletedTask;
+                    },
+                    OnTokenValidated = c =>
+                    {
+                        c.Principal.AddIdentity(new ClaimsIdentity(new Claim[]
+                        {
+                            new Claim(SecurityConstants.AuthLevelClaimType, AuthorizationLevel.Admin.ToString())
+                        }));
 
-                                    c.Success();
+                        c.Success();
 
-                                    return Task.CompletedTask;
-                                }
-                            };
+                        return Task.CompletedTask;
+                    }
+                };
 
-                            o.TokenValidationParameters = CreateTokenValidationParameters();
+                o.TokenValidationParameters = CreateTokenValidationParameters(GetValidAudiences());
+
+                // TODO: DI (FACAVAL) Remove this once the work above is completed.
+                if (!SystemEnvironment.Instance.IsPlaceholderModeEnabled())
+                {
+                    // We're not in standby mode, so flag as specialized
+                    _specialized = 1;
+                }
+            });
+
+        private static string[] GetValidAudiences()
+        {
+            if (SystemEnvironment.Instance.IsPlaceholderModeEnabled() &&
+                SystemEnvironment.Instance.IsLinuxConsumptionOnLegion())
+            {
+                return new string[]
+                {
+                    ScriptSettingsManager.Instance.GetSetting(WebsitePodName)
```

Comment:
Hmmm, wondering if we shouldn't use the SiteUriFormat as below in this scenario for consistency. Using this pod name instead of the site name ""https://<podname>.azurewebsites.net".

Reply:
I thought about this too. Althought I didn't have too much of a preference, I settled for this because

- No need to do `string.Format` when we don't have to.
- Other containers (existing and new) can just take things from `WEBSITE_POD_NAME`.
-  Simplifies things both in terms of implementation and explanation. (Imagine having to explain , why do you suffix ".azurewebsites.net" when it's not a real site?)

Reply:
Use of a non URI value is to spec https://www.rfc-editor.org/rfc/rfc7519 as you're doing is to spec - the value just needs to be a StringOrUri. You're arguing that because the container isn't specialized yet, an App Service site doesn't really exist - it's just a non-externally addressable container. Fine I guess.